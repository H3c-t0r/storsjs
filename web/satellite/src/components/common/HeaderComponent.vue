// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

<template>
    <div class="header-container">
        <div class="header-container__buttons-area">
            <slot></slot>
        </div>
        <SearchComponent ref="search" :placeHolder="placeHolder" :search="search"/>
    </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';

import SearchComponent from '@/components/common/SearchComponent.vue';

declare type searchCallback = (search: string) => Promise<void>;
declare interface ClearSearch {
    clearSearch(): void;
}

@Component({
    components: {
        SearchComponent,
    },
})
export default class HeaderComponent extends Vue {
    @Prop({default: ''})
    private readonly placeHolder: string;
    @Prop({default: () => ''})
    private readonly search: searchCallback;

    public $refs!: {
        search: SearchComponent & ClearSearch;
    };

    public clearSearch() {
        this.$refs.search.clearSearch();
    }
}
</script>

<style scoped lang="scss">
    .header-container {
        width: 100%;
        height: 85px;
        position: relative;
        display: flex;
        align-items: center;
        justify-content: space-between;

        &__buttons-area {
            width: auto;
            display: flex;
            align-items: center;
            justify-content: space-between;
        }
    }
</style>
