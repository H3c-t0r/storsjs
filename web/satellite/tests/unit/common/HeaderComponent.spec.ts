// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

import { mount, shallowMount } from '@vue/test-utils';
import * as sinon from 'sinon';
import HeaderComponent from '@/components/common/HeaderComponent.vue';

describe('HeaderComponent.vue', () => {
    it('renders correctly', () => {
        const wrapper = shallowMount(HeaderComponent);

        expect(wrapper).toMatchSnapshot();
    });

    it('renders correctly with default props', () => {
        const wrapper = mount(HeaderComponent);

        expect(wrapper.vm.$props.placeHolder).toMatch('');
        expect(wrapper.vm.$props.search).toMatch('');
        expect(wrapper.vm.$props.title).toMatch('');
    });

    it('function clearSearch works correctly', () => {
        let clearSearchSpy = sinon.spy();

        const wrapper = mount(HeaderComponent);

        wrapper.vm.$refs.search.clearSearch = clearSearchSpy;

        wrapper.vm.clearSearch();

        expect(clearSearchSpy.callCount).toBe(1);
    });
});
