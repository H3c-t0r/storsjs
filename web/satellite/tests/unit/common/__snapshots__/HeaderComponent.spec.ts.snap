// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderComponent.vue renders correctly 1`] = `
<div class="header-container">
  <h1></h1>
  <div class="header-container__wrapper">
    <div class="header-container__buttons-area"></div>
    <searchcomponent-stub placeholder="" search=""></searchcomponent-stub>
  </div>
</div>
`;
